docs/handling-opentelemetry-references.md

```diff
@@ -0,0 +1,66 @@
+# Handling OpenTelemetry SDK/Instrumentation References in Applications with Auto-Instrumentation
```

Comment:
Some time ago we have decided to avoid "auto-instrumentation" in our doc. Please fix all places here

```suggestion
# Handling OpenTelemetry SDK/Instrumentation References in Applications with Automatic Instrumentation
```